src/libraries/Native/AnyOS/System.Globalization.Native/pal_collation.c

```diff
@@ -403,8 +407,12 @@ static const UCollator* GetCollatorFromSortHandle(SortHandle* pSortHandle, int32
         pCollator = CloneCollatorWithOptions(pSortHandle->collatorsPerOption[0], options, pErr);
         UCollator* pNull = NULL;
 
+#ifdef TARGET_UNIX
         // we are not using the standard atomic_compare_exchange_strong to workaround bugs in clang 5.0 (https://bugs.llvm.org/show_bug.cgi?id=37457)
```

Comment:
Do we still need this workaround? I believe we are not building with clang5 anymore.

Reply:
Yeah. We now use clang9 so we can remove this workaround. I will fix it as part of this.

Reply:
If I try to use `atomic_compare_exchange_strong` I get an error: 
>/home/runtime/src/libraries/Native/Unix/System.Globalization.Native/pal_collation.c:410:46: error: _Atomic cannot be applied to incomplete type 'UCollator' (aka 'struct UCollator')
        if (!atomic_compare_exchange_strong((_Atomic(UCollator) *)&pSortHandle->collatorsPerOption[options], &pNull, pCollator))
> /usr/include/unicode/ucol.h:58:8: note: forward declaration of 'struct UCollator'
struct UCollator;

It seems like the declaration of UCollator in the icu headers is incomplete.

Reply:
maybe you need to ensure 

#include "unicode/uloc.h"

is included?

Reply:
It was included. I managed to do it with `void* volatile*`. I'll put an update soon.

Reply:
It seems like in gcc `atomic_compare_exchange_strong` uses `__atomic_compare_exchange` macro which expects other parameter types, if I instead directly use `__atomic_compare_exchange_n` in gcc builds just fine. Will leave as is in order to have a green build at the moment. 

> /__w/1/s/src/libraries/Native/Unix/Common/pal_atomic.h:15:12: error: size mismatch in argument 2 of '__atomic_compare_exchange'
     return atomic_compare_exchange_strong_explicit((_Atomic(void*)volatile*)dest, exchange, comperand,  __ATOMIC_SEQ_CST, __ATOMIC_SEQ_CST);